src/Cache/FileHandler.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Cache;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+
+final class FileHandler implements FileHandlerInterface
+{
+    private string $file;
+
+    public function __construct(string $file)
+    {
+        $this->file = $file;
+    }
+
+    public function getFile(): string
+    {
+        return $this->file;
+    }
+
+    public function read(): ?CacheInterface
+    {
+        if (!file_exists($this->file)) {
+            return null;
+        }
+
+        $content = file_get_contents($this->file);
+        if (false === $content) {
+            return null;
+        }
+
+        return Cache::fromJson($content);
+    }
+
+    /**
+     * @throws IOException
+     */
+    public function write(CacheInterface $cache): void
+    {
+        $content = $cache->toJson();
```

Comment:
There is something wrong here.

We're calling `$cache->toJson();`, using the implementation of toJson of the CacheInterface which can be custom.

But we're using `Cache::fromJson($content)` which is our implementation so might not work with a custom implementation.

Instead of providing a `toJson` method in the CacheInterface, we would need to expose `getHashes` or `getAll` (or a better name) so we might create a service with the toJson/fromJson method (or implement them directly in the FileHandler).